patterns/guaranteed_receiver_nack.go

```diff
@@ -0,0 +1,128 @@
+		WithRequiredMessageOutcomeSupport(config.PersistentReceiverFailedOutcome, config.PersistentReceiverRejectedOutcome).
+		Build(durableExclusiveQueue)
+
+	// // Build a Gauranteed message receiver with NACK support and bind to the given queue
+	// // Set the required message settlement outcome(s) on the flow using the configuration provider
+	// persistentReceiver, err := messagingService.CreatePersistentMessageReceiverBuilder().
+	// 	WithMessageClientAcknowledgement().
+	// 	// Add message settlement outcomes support on the created Flow here (Failed and Reject for NACK(ing) messages)
+	// 	FromConfigurationProvider(config.ReceiverPropertyMap{
+	// 		config.ReceiverPropertyPersistentMessageRequiredOutcomeSupport: fmt.Sprintf("%s,%s",config.PersistentReceiverFailedOutcome, config.PersistentReceiverRejectedOutcome),
+	// 	}).
+	// 	Build(durableExclusiveQueue)
+
+	// Handling a panic from a non existing queue
+	defer func() {
+		if err := recover(); err != nil {
+			fmt.Printf("Make sure queue name '%s' exists on the broker.\nThe following error occurred when attempting to connect to create a Persistent Message Receiver:\n%s", queueName, err)
+		}
+	}()
+
+	// Start Persistent Message Receiver
+	if err := persistentReceiver.Start(); err != nil {
+		panic(err)
+	}
+
+	fmt.Println("Persistent Receiver running? ", persistentReceiver.IsRunning())
+
+	// Message Handler
+	messageHandler := func(message message.InboundMessage) {
+		var messageBody string
+
+		if payload, ok := message.GetPayloadAsString(); ok {
+			messageBody = payload
+		} else if payload, ok := message.GetPayloadAsBytes(); ok {
+			messageBody = string(payload)
+		}
+
+		fmt.Printf("Received Message Body %s \n", messageBody)
+		// fmt.Printf("Message Dump %s \n", message)
+
+		// Settle the message here with one of the three supported settlement outcomes: ACCEPTED, FAILED and REJECTED
+		messageSettlementError := persistentReceiver.Settle(message, config.PersistentReceiverAcceptedOutcome) // Accept(acknowlegde) the message
+		// messageSettlementError := persistentReceiver.Settle(message, config.PersistentReceiverFailedOutcome) // fail the message
+		// messageSettlementError := persistentReceiver.Settle(message, config.PersistentReceiverRejectedOutcome) // reject the message
+		fmt.Println("Message Settlement Error: ", messageSettlementError)
+	}
+
+	// Register Message callback handler to the Message Receiver
+	if regErr := persistentReceiver.ReceiveAsync(messageHandler); regErr != nil {
+		panic(regErr)
+	}
+	fmt.Printf("\n Bound to queue: %s\n", queueName)
+	fmt.Println("\n===Interrupt (CTR+C) to handle graceful termination of the receiver===\n")
+
+	// Run forever until an interrupt signal is received
+	// Handle interrupts
+
+	c := make(chan os.Signal, 1)
+	signal.Notify(c, os.Interrupt)
+
+	// Block until a signal is received.
+	<-c
+
+	// Terminate the Persistent Receiver
+	persistentReceiver.Terminate(1 * time.Second)
```

Comment:
@oodigie , you need to add a comment justifying the timeout value that you gave so that the user understands whether it was arbitrary, or if a 1 second timeout is necessary/recommended for this use case.

Reply:
@TrentDaniel, I don't think this is really necessary, please see the other samples. The Go PubSub+ API documentation clearly states what a grace period is and there is no explanation as to why we use 1 second or 10 seconds. The selected grace period is at the discretion of the dev.

Reply:
@oodigie , if a new customer is trying to learn about how to use our product or this feature, they could be confused as to why a specific grace period is given. They might wonder if 1 second is passively being recommended as the minimum grace period on termination for this feature, since it is not configurable through the sample and it there is no explanation. If this is done elsewhere in the samples, then those are just additional examples of ambiguity, and do not justify this instance of ambiguity. Please either state in the comment that the timeout is arbitrary, or that it was selected as 1 second for a specific reason.